### ISSUE TITLE
`manual_unwrap_or` should not trigger inside const fn

### ISSUE THREAD

**Issue by davidhewitt** [rust-lang/rust-clippy#6898]
I tried this code:

```rust
const fn baz(x: Option<i32>) -> i32 {
    match x {
        Some(x) => x,
        None => 0
    }
}
```

I expected to see this happen: no lint error, beacuse `.unwrap_or` cannot be used inside a `const fn` yet.

Instead, this happened: `manual_unwrap_or` lint triggers.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (cb75ad5 2021-02-10)
- `rustc -Vv`:
  ```
  rustc 1.50.0 (cb75ad5db 2021-02-10)
  binary: rustc
  commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
  commit-date: 2021-02-10
  host: x86_64-unknown-linux-gnu
  release: 1.50.0
  ```
